### PR TITLE
Microsoft LINK personality

### PR DESCRIPTION
**Pull request by kroening**

### CODE REVIEW

File: src/goto-cc/ms_link_mode.cpp [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+  // Don't add the system paths!
+
+  if(cmdline.isset("OUT"))
+  {
+    compiler.output_file_executable = cmdline.get_values("OUT").back();
+  }
+  else
+  {
+    compiler.output_file_object = "";
+    compiler.output_file_executable = "a.out";
```
Comment (tautschnig) [inline src/goto-cc/ms_link_mode.cpp]: Is this really the default for link?

**Review by tautschnig — APPROVED** [review]

File: src/goto-cc/ms_link_mode.cpp (hunk 1 above)
Comment (kroening) [reply src/goto-cc/ms_link_mode.cpp]: No, fixed.

**Review by allredj — COMMENTED** [review]
This PR failed Diffblue compatibility checks (cbmc commit: 1688679937ed2266eeb6f08ec9721b99b473ddeb).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/80883566
Status will be re-evaluated on next push.
Please contact @peterschrammel, @thk123, or @allredj for support.

Common spurious failures:
  - the cbmc commit has disappeared in the mean time (e.g. in a force-push)
  - the author is not in the list of contributors (e.g. first-time contributors).
##

**Comment by thk123** [comment]
Unless I am missing something here - surely this requires some tests?

**Comment by tautschnig** [comment]
Once #2673 is in, adding tests should be doable.

**Review by allredj — COMMENTED** [review]
Passed Diffblue compatibility checks (cbmc commit: 1496b3ee80b7ea68bfbc9af5ec33e6dd9d61b152).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/80887339

**Review by allredj — COMMENTED** [review]
Passed Diffblue compatibility checks (cbmc commit: ded18c46f33de5ac4a7f394d8db0e3f56fbdc5e3).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/80945427

**Comment by tautschnig** [comment]
Test added - now testing the test ...

**Review by allredj — COMMENTED** [review]
Passed Diffblue compatibility checks (cbmc commit: 7cd63995a64e261a176867fbc7c7947bbb9a9462).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/81020735

**Review by allredj — COMMENTED** [review]
Passed Diffblue compatibility checks (cbmc commit: 88591d8e8f8c6ecd048276fd5c324acabcc0665e).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/81034798

**Review by allredj — COMMENTED** [review]
Passed Diffblue compatibility checks (cbmc commit: 0606aeff90341095342b4a36395bc413fca94baf).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/81043455

**Comment by tautschnig** [comment]
This will need #2688 to be merged before CI can pass.

**Review by allredj — COMMENTED** [review]
Passed Diffblue compatibility checks (cbmc commit: 8530232c91fbcd6886a2ae0e1818e710be2a71de).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/81060405

**Comment by tautschnig** [comment]
... and #2687 as well.

**Review by allredj — COMMENTED** [review]
Passed Diffblue compatibility checks (cbmc commit: de075db7392ea393dc4034a6383cd4d72bd1a77f).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/81065000

**Review by allredj — COMMENTED** [review]
Passed Diffblue compatibility checks (cbmc commit: c63030ffa7fb51342d708b997631160c56946d7c).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/81074442

**Pull request merged**